src/openfermion/transforms/_code_transform_functions.py

```diff
@@ -0,0 +1,254 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+""" Pre-existing codes for Fermion-qubit mappings
+    based on (arXiv:1712.07067) """
+
+import numpy
+
+from openfermion.ops import BinaryCode, SymbolicBinary, linearize_decoder
+
+
+def _encoder_bk(dimension):
+    """  Helper function for bravyi_kitaev_code that outputs the binary-tree
+    (dimension x dimension)-matrix used for the encoder in the
+    Bravyi-Kitaev transform.
+
+    Args:
+        dimension (int): length of the matrix, the dimension x dimension
+
+    Returns (numpy.ndarray): encoder matrix
+    """
+    reps = int(numpy.ceil(numpy.log2(dimension)))
+    mtx = numpy.array([[1, 0], [1, 1]])
+    for repetition in numpy.arange(1, reps + 1):
+        mtx = numpy.kron(numpy.eye(2, dtype=int), mtx)
+        for column in numpy.arange(0, 2 ** repetition):
+            mtx[2 ** (repetition + 1) - 1, column] = 1
+    return mtx[0:dimension, 0:dimension]
+
+
+def _decoder_bk(modes):
```

Comment:
I would recommend calling this n_modes to be consistent with the sort of notation we use elsewhere.